plugins/modules/pacemaker_stonith.py

```diff
@@ -0,0 +1,228 @@
+        type: str
+      operation_options:
+        description:
+          - Operation option to associate with action.
+        type: list
+        elements: str
+  stonith_metas:
+    description:
+      - List of meta to associate with STONITH.
+    type: list
+    elements: str
+  stonith_argument:
+    description:
+      - Action to associate with STONITH.
+    type: dict
+    suboptions:
+      argument_action:
+        description:
+          - Action to apply to STONITH.
+        type: str
+        choices: [ group, before, after ]
+      argument_options:
+        description:
+          - Options to associate with STONITH action.
+        type: list
+        elements: str
+  agent_validation:
+    description:
+      - enabled agent validation for STONITH creation.
+    type: bool
+    default: false
+  wait:
+    description:
+      - Timeout period for polling the STONITH creation.
+    type: int
+    default: 300
+'''
+
+EXAMPLES = '''
+---
+- name: Create pacemaker STONITH
+  hosts: localhost
+  gather_facts: false
+  tasks:
+  - name: Create virtual-ip STONITH
+    community.general.pacemaker_stonith:
+      state: present
+      name: virtual-stonith
+      stonith_type: fence_virt
+      stonith_options:
+        - "pcmk_host_list=f1"
+      stonith_operations:
+        - operation_action: monitor
+          operation_options:
+            - "interval=30s"
+'''
+
+RETURN = '''
+cluster_stonith:
+    description: The cluster STONITH output message.
+    type: str
+    sample: ""
+    returned: always
+'''
+
+from ansible_collections.community.general.plugins.module_utils.module_helper import StateModuleHelper
+from ansible_collections.community.general.plugins.module_utils.pacemaker import pacemaker_runner
+
+
+class PacemakerStonith(StateModuleHelper):
+    module = dict(
+        argument_spec=dict(
+            state=dict(type='str', default='present', choices=[
+                'present', 'absent', 'enabled', 'disabled']),
+            name=dict(type='str', required=True),
+            stonith_type=dict(type='str'),
+            stonith_options=dict(type='list', elements='str', default=list()),
+            stonith_operations=dict(type='list', elements='dict', default=list(), options=dict(
+                operation_action=dict(type='str'),
+                operation_options=dict(type='list', elements='str'),
+            )),
+            stonith_metas=dict(type='list', elements='str'),
+            stonith_argument=dict(type='dict', options=dict(
+                argument_action=dict(type='str', choices=['before', 'after', 'group']),
+                argument_options=dict(type='list', elements='str'),
+            )),
+            agent_validation=dict(type='bool', default=False),
+            wait=dict(type='int', default=300),
+        ),
+        required_if=[('state', 'present', ['stonith_type', 'stonith_options'])],
+        supports_check_mode=True
+    )
+
+    use_old_vardict = False
+    default_state = "present"
```

Comment:
Since `state` will always have some value, and its default is `present`, this is kinda redundant.

For the record - I just noticed this and have made that same mistake in many of my own modules ;-)